packages/payload/src/admin/components/utilities/Preferences/index.tsx

```diff
@@ -65,14 +67,57 @@ export const PreferencesProvider: React.FC<{ children?: React.ReactNode }> = ({
   )
 
   const setPreference = useCallback(
-    async (key: string, value: unknown): Promise<void> => {
-      preferencesRef.current[key] = value
-      await requests.post(
-        `${serverURL}${api}/payload-preferences/${key}`,
-        requestOptions(value, i18n.language),
-      )
+    async (key: string, value: unknown, merge = false): Promise<void> => {
```

Comment:
Some comments might be helpful with all the if statements in here

Reply:
I commented up top in the PreferencesContext type.